src/test/java/com/alvarium/annotators/http/ParseResultTest.java

```diff
@@ -0,0 +1,181 @@
+/*******************************************************************************
+ * Copyright 2022 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package com.alvarium.annotators.http;
+
+import org.apache.http.client.methods.HttpPost;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertEquals;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public class ParseResultTest {
+  
+  HttpPost getRequest() {
+    HttpPost request = new HttpPost(URI.create("http://example.com/foo?var1=&var2=2"));
+                request.setHeader("Host", "example.com");
+                request.setHeader("Date", "Tue, 20 Apr 2021 02:07:55 GMT");
+                request.setHeader("Content-Type", "application/json");
+                request.setHeader("Content-Length", "18");
+                request.setHeader("Signature-Input", "");
+                request.setHeader("Signature", "whatever");
+    return request;
+  }
+
+  @Rule
+  public ExpectedException exceptionRule = ExpectedException.none();
+  
+  @Test
+  public void testIntegrationAllFields() throws URISyntaxException, ParseResultException{
```

Comment:
I notice `@target-uri` isn't included in this test but it probably should be.

Reply:
Fixed in [dcd31ee](https://github.com/project-alvarium/alvarium-sdk-java/commit/dcd31ee4a3869875e07ccf0c28a96420e26ec365)